src/Tree/Mapping/Driver/Annotation.php

```diff
@@ -180,7 +180,7 @@ public function readExtendedMetadata($meta, array &$config)
                 $config['root'] = $field;
             }
             // level
-            if ($this->reader->getPropertyAnnotation($property, self::LEVEL)) {
+            if ($levelAnnotation = $this->reader->getPropertyAnnotation($property, self::LEVEL)) {
```

Comment:
Do you mean to separate the assignment with the condition on 2 separate lines?
Generally I've tried to keep to the existing code style and you can see the following `if` on line 195 is done the same way.

Reply:
Yes, but I haven't a strong argument here. Generally, the assignment inside a condition leads more changes in case of refactoring. Additionally, it makes harder to annotate the variables (when it is strictly required).